src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -24045,6 +24069,88 @@ void Init()
                 );
         }
 
+        [Fact]
+        public void ConstructorUsesStateFromInitializers()
+        {
+            var source = @"
+public class Program
+{
+    public string Prop { get; set; } = ""a"";
+
+    public Program()
+    {
+        Prop.ToString();
+    }
+}
+";
+            var comp = CreateNullableCompilation(new[] { source });
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void ConstMembersUseDeclaredNullability_01()
+        {
+            var source = @"
+public class Program
+{
+    const string s1 = ""hello"";
+    public static readonly string s2 = s1.ToString();
+
+    public Program()
+    {
+    }
+}
+";
+            var comp = CreateNullableCompilation(new[] { source });
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void ConstMembersUseDeclaredNullability_02()
+        {
+            var source = @"
+public class Program
+{
+    const string? s1 = ""hello"";
+
+    public Program()
+    {
+        var x = s1.ToString(); // 1
+    }
+}
+";
+            // Arguably we could just see through to the constant value and know it's non-null,
+            // but it feels like the user should just fix the type of their 'const' in this scenario.
```

Comment:
It feels very weird that this would give a warning, but this does not:
```cs
public class Program
{
    public string? Prop { get; set; } = ""a"";
    public Program()
    {
        Prop.ToString();
    }
}
```

Reply:
Also, it's certainly possible that the user _can't_ change the type, because in some `#if`s it's actually null.

---
In reply to: [470284592](https://github.com/dotnet/roslyn/pull/46441#discussion_r470284592) [](ancestors = 470284592)

Reply:
Opened a follow-up issue #46805